src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackageDependenciesAndToolsRule.cs

```diff
@@ -67,12 +75,57 @@ public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migra
                     migrationRuleInputs.OutputMSBuildProject,
                     targetFramework.FrameworkName, 
                     targetFramework.Dependencies,
-                    migrationRuleInputs.ProjectXproj); 
+                    migrationRuleInputs.ProjectXproj);
+
+                if (defaultProjectTargetFramework.IsDesktop())
+                {
+                    allDependencies.AddRange(targetFramework.Dependencies);
+                }
+            }
+
+            // Automatically add the same desktop references that the build for project.json does.
+            // This ensures if the project.json build succeeded then the migrated msbuild one 
+            // will succeed too. See src\Microsoft.DotNet.ProjectModel\Resolution for more details.
+            if (defaultProjectTargetFramework.IsDesktop())
+            {
+                List<ProjectLibraryDependency> autoAddDesktopDependencies = new List<ProjectLibraryDependency>();
+
+                // MSBuild targets automatically add a reference to mscorlib
+                // AddIfMissing(autoAddDesktopDependencies, allDependencies, "mscorlib");
+                AddIfMissing(autoAddDesktopDependencies, allDependencies, "System");
+
+                // MSBuild targets automatically add a reference to System.Core
+                //if (defaultProjectTargetFramework.Version >= new Version(3, 5))
```

Comment:
I wanted to document that I purposely left these out because msbuild magically adds them. I agree with what you said, but in this case felt it made sense? But if you prefer I'll remove them.